# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/HttpConnectionUDP.cpp
# Commit: 521c9e019678
# Full Hash: 521c9e019678f584590f63473416d40e03f33342
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2023-03-15 04:10:55
# Regressor Bug: 1805372
# File Overlap Count: 1
# Description:
#   Bug 1805372 - Pass effectiveTRRMode to nsHttpChannel from socketTransport/DnsAndConnectSocket r=necko-reviewers,kershaw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D164539
# ==============================================================================

diff -r dfb39ca05529 -r 521c9e019678 netwerk/protocol/http/HttpConnectionUDP.cpp
--- a/netwerk/protocol/http/HttpConnectionUDP.cpp	Tue Mar 14 15:29:49 2023 +0200
+++ b/netwerk/protocol/http/HttpConnectionUDP.cpp	Tue Mar 14 12:59:40 2023 +0000
@@ -77,6 +77,8 @@
     return NS_ERROR_FAILURE;
   }
   dnsAddrRecord->IsTRR(&mResolvedByTRR);
+  dnsAddrRecord->GetEffectiveTRRMode(&mEffectiveTRRMode);
+  dnsAddrRecord->GetTrrSkipReason(&mTRRSkipReason);
   NetAddr peerAddr;
   nsresult rv = dnsAddrRecord->GetNextAddr(mConnInfo->GetRoutedHost().IsEmpty()
                                                ? mConnInfo->OriginPort()
@@ -675,5 +677,11 @@
 
 bool HttpConnectionUDP::ResolvedByTRR() { return mResolvedByTRR; }
 
+nsIRequest::TRRMode HttpConnectionUDP::EffectiveTRRMode() {
+  return mEffectiveTRRMode;
+}
+
+TRRSkippedReason HttpConnectionUDP::TRRSkipReason() { return mTRRSkipReason; }
+
 }  // namespace net
 }  // namespace mozilla